### PR TITLE
meta: ci & linting configuration

### PR DESCRIPTION
**Pull request by simbuerg**
This pull request combines a set of modifications to our CI/Linting configuration:

* Format string literals and require at least python 3.7, so we drop 3.5 and 3.6.
* Enable python 3.8 on travis.
* Small "fixes" to pylint, to make it less annoying for now.
  This disables comments about undocumented methods/classes.
  *It is still the goal to have them all commented properly.*
  A future PR can deal with that.
* Disable betttercodehub: Wasn't that useful, but very annoying. Almost the same featureset as 
  codacy, so we stick to codacy.
* Improve coverage reporting: Remove duplicates and further path adjustments to get meaningful 
  coverage information.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,17 +21,23 @@ install:
...
-  - coverage run -a `which benchbuild` -vvvvv run --full test
+  - coverage run -p `which benchbuild` bootstrap -s
+  - pytest --cov-config=./.coveragerc --cov=./ benchbuild tests
+  - coverage run -p `which benchbuild` config view
+  - coverage run -p `which benchbuild` project view
+  - coverage run -p `which benchbuild` experiment view
+  - coverage run -p `which benchbuild` -vvvvv run --full test
+  - coverage combine -a
+  - coverage report
+  - mypy --strict benchbuild
```
Comment (vulder) [inline .travis.yml:38]: that works on full BB?

File: .travis.yml [hunk 2]
```diff
@@ -22,17 +21,23 @@ install:
...
   - BB_UNIONFS_ENABLE=false BB_PLUGINS_EXPERIMENTS='["benchbuild.experiments.raw", "benchbuild.experiments.empty"]'
 
 script:
-  - coverage run -a `which benchbuild` bootstrap -s
-  - pytest --cov=./ benchbuild tests
-  - coverage run -a `which benchbuild` config view
-  - coverage run -a `which benchbuild` project view
-  - coverage run -a `which benchbuild` experiment view
-  - coverage run -a `which benchbuild` -vvvvv run --full test
+  - coverage run -p `which benchbuild` bootstrap -s
```
Comment (vulder) [inline .travis.yml:30]:
```suggestion
  - pre-commit run --from-ref origin/master --to-ref HEAD
  - coverage run -p `which benchbuild` bootstrap -s
```

If this passes for BB :wink:

Comment (simbuerg) [reply .travis.yml:30]: pre-commit is not active on master yet

File: .travis.yml (hunk 1 above)
Comment (simbuerg) [reply .travis.yml:38]: mypy? No. Errors out like crazy. Mhh you're right, maybe i just let it run but don't fail the build.
I'll have another look at this.

Comment (vulder) [reply .travis.yml:38]: You can also try non strict mode or tell mypy to only check the diff, but both alternatives assume you have some level of type annotations in the code base.

File: .travis.yml (hunk 2 above)
Comment (vulder) [reply .travis.yml:30]: It does not have to be, this just tells pre-commit only to check the diff between the current HEAD and master, i.e., only complaining if the PR has bad patterns but not if the rest of the code base has them.

Comment (simbuerg) [reply .travis.yml:30]: This requires pre-commit to be available in the travis build environment. So either it is installed in .travis.yml or it is provided by requirements.txt

Neither is true.

Comment (vulder) [reply .travis.yml:30]: kk than add it later

**Comment by simbuerg** [comment]
rebase against master

**Pull request merged**